File: src/main/java/wanted/Market/domain/product/dto/ProductResponse.java
package wanted.Market.domain.product.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import wanted.Market.domain.product.entity.ProductStatus;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class ProductResponse {
    private String name;
    private int price;
    private ProductStatus status;
}

File: src/main/kotlin/wanted/market/product/ProductResponse.kt
package wanted.market.product

data class ProductResponse(val name: String,
                           val price: Int,
                           val productStatus: String
){
    companion object {
        fun from(product: Product): ProductResponse {
            return ProductResponse(product.name, product.price, product.productStatus.value)
        }
    }
}

